File: Interview 40. getLeastNumbers/getLeastNumbers.py
class getLeastNumbers:
    def solution(self, arr, k):
        count = [0]*10000
        for i in arr:
            count[i] = count[i] + 1
        sortedArr = []
        for idx, item in enumerate(count):
            while item:
                sortedArr.append(idx)
                item = item - 1
        return sortedArr[0:k]

obj = getLeastNumbers()
arr = [0,1,2,1]
k = 2
res = obj.solution(arr, k)
print(res)

File: Interview 40. getLeastNumbers/getLeastNumbers.java
import java.util.*;

class getLeastNumbers {
    public int[] solution(int[] arr, int k) {
    	int[] count = new int[10000];
    	for (int i = 0; i < arr.length; i++) {
    		count[arr[i]] = count[arr[i]] + 1;
    	}
    	int[] sortedArr = new int[arr.length];
    	int index = 0;
    	for (int i = 0; i < arr.length; i++) {
    		while (count[i] > 0) {
    			sortedArr[index] = i;
    			index = index + 1;
    			count[i] = count[i] - 1;
    		}
    	}
    	return Arrays.copyOfRange(sortedArr, 0, k);
    }

    public static void main(String[] argv) {
    	getLeastNumbers obj = new getLeastNumbers();
    	int[] arr = {0,1,2,1};
    	int k = 2;
    	int[] res = obj.solution(arr, k);
    	for (int i = 0; i < res.length; i++) {
    		System.out.print(res[i] + " ");
    	}
    	System.out.println(" ");
    }
}